app_feup/lib/model/entities/time_utilities.dart

```diff
@@ -1,7 +1,39 @@
+import 'package:intl/intl.dart';
+
+const weekdays = [
+  'N/A',
+  'Segunda',
+  'Terça',
+  'Quarta',
+  'Quinta',
+  'Sexta',
+  'Sábado',
+  'Domingo'
+];
+
 extension TimeString on DateTime {
   String toTimeHourMinString() {
-    return this.hour.toString().padLeft(2, '0') +
-        ':' +
-        this.minute.toString().padLeft(2, '0');
+    return DateFormat('kk:mm').format(this);
   }
-}
\ No newline at end of file
+
+  String formatter() {
+    final now = DateTime.now();
+
+    final daysDif = now.difference(this).inDays;
+
+    if (0 < daysDif && daysDif < 1) {
+      final hoursDif = now.difference(this).inHours; // hours difference
+
+      if (hoursDif == 0) {
+        final minutesDif = now.difference(this).inMinutes; // minutes difference
+        return 'Há $minutesDif minutos';
+      }
+      return 'Há $hoursDif horas';
+    } else if (daysDif == 1) {
+      return 'Ontem';
+    } else if (0 < daysDif && daysDif < 7) {
+      return weekdays[this.weekday];
+    }
+    return DateFormat('dd/MM').format(this); //yyyy-MM-dd – kk:mm
+  }
+}
```

Comment:
Can we rename this to `toFormattedDateString` to make it coherent with the already created utility?
Also, the `inHours` method from the `Duration` class returns an int. Does the first check make sense?

```suggestion
  String toFormattedDateString() {
    final now = DateTime.now();
    final daysDif = now
        .difference(this)
        .inDays;

    if (daysDif == 0) {
      final hoursDif = now.difference(this).inHours;
      if (hoursDif == 0){
        final minutesDif = now.difference(this).inMinutes;
        return 'Há $minutesDif minutos';
      }
      return 'Há $hoursDif horas';
    }
    if (daysDif == 1) {
      return 'Ontem';
    }
    if (daysDif > 0 && daysDif < 7) {
      return weekdays[this.weekday];
    }
    return DateFormat('dd/MM').format(this); //yyyy-MM-dd – kk:mm
  }
```